templates/vue-starter-template/app/app.vue

```diff
@@ -1,3 +1,14 @@
+<script setup lang="ts">
+import type { Schemas } from "#shopware";
+
+const { apiClient } = useShopwareContext();
+const sessionContextData = ref<Schemas["SalesChannelContext"]>();
+const contextResponse = await apiClient.invoke("readContext get /context");
```

Comment:
do we want to stop rendering for context fetching?

1. it should probably be CSR-only action
2. would be nice to make it more asynchronous

if 2. does not make sense then maybe it should be in asyncData then?

Reply:
I don't know the context of the change here, https://github.com/shopware/frontends/pull/309/files#diff-4359a385df4e4b0363f4e54f8cc7240a106ecfcf1be47ec09b91f7537990805fL20, but it was `asycData` before, and changed by you.  

Moving request to CSR will break the app because the session context will  not be set on the SSR

Reply:
yeah, the reason was to not share session token from server-cached render, that's why my first suggestion was to make it CSR-only

we should break down what exactly will break and maybe fix it? Would be nice to have server render session-independent. Additionally not to wait with CSR render/hydration with fetching session, but rather async load it

Reply:
I created separated task for this https://github.com/shopware/frontends/issues/1844